Automattic/jetpack
# Connection: Update the connection initial state to fallback on the new consolidated Jetpack script data

In #38430, we worked on creating a consolidated initial state (called script data) to use everywhere in Jetpack to avoid its duplication all over the place as discussed in pdWQjU-KI-p2

This is needed to continue working on #38330.

## Proposed changes:

* Add connection initial state to `window.JetpackScriptData`
* Keep the existing initial state untouched
* Update the JS components to fallback on the new consolidated script data to allow gradual migration

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Goto any Jetpack page that relies on connection, e.g. `/wp-admin/admin.php?page=jetpack#/settings`
* Confirm that everything works same as before
* Open the browser console
* Confirm that `window.JetpackScriptData.connection` is same as `JP_CONNECTION_INITIAL_STATE`
